staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/metrics/metrics.go

```diff
@@ -209,6 +221,10 @@ func RecordKeyIDFromStatus(providerName, keyID string) {
 	KeyIDHashStatusLastTimestampSeconds.WithLabelValues(providerName, keyIDHash).SetToCurrentTime()
 }
 
+func RecordInvalidKeyIDFromStatus(providerName string, err error) {
+	InvalidKeyIDFromStatusTotal.WithLabelValues(providerName, err.Error()).Inc()
```

Comment:
Unbounded label value on the `err.Error()`?  I think we need to add some logic to make sure that code changes to the validation function do not cause changes in the metrics.

Reply:
I was thinking about this too. I think a function to return static errors based on the error message from `ValidateKeyID` will take care of the unbounded label issue. Or we could just come up with errors codes `key_id_empty`, `key_id_too_long` and `unknown_error` similar to the grpc metrics we're doing.

Reply:
How about we set few const for err messages to use in ValidateKeyID and if the errors are outside of that list, then RecordInvalidKeyIDFromStatus records an item with error like "error is not a predefined error"?

Reply:
At a high level I think we need to have both the concept of an "error code" (for use in metrics) and the "error message" (this is the `.Error` we already have).